### PR TITLE
Adding support for polygon intersects method and creation from GeoJSON

### PR DESCRIPTION
**Pull request by CodingMarmot**
- Adding NewGeometryFromGeoJSON function
- Adding Geometry.Intersects function

### CODE REVIEW

File: godal.cpp [hunk 1]
```diff
@@ -685,6 +685,13 @@ OGRGeometryH godal_OGR_G_Buffer(cctx *ctx, OGRGeometryH in, double tolerance, in
 	return ret;
 }
 
+int godal_OGR_G_Intersects(cctx *ctx, OGRGeometryH geom1, OGRGeometryH geom2) {
+	godalWrap(ctx);
```
Comment (tbonfort) [inline godal.cpp:689]: Error checking/handling should be done in the same way as in `godal_OGR_G_Buffer`

File: godal.cpp [hunk 2]
```diff
@@ -848,6 +855,21 @@ void godalClearRasterStatistics(cctx *ctx, GDALDatasetH ds){
   godalUnwrap();
 }
 
+OGRGeometryH godalNewGeometryFromGeoJSON(cctx *ctx, char *geoJSON) {
+	godalWrap(ctx);
+	OGRGeometryH gptr = OGR_G_CreateGeometryFromJson(geoJSON);
+	if (gptr == nullptr) {
+		fprintf(stderr,"error occurred\n");
```
Comment (tbonfort) [inline godal.cpp:862]: remove debugging line

File: options.go [hunk 3]
```diff
@@ -410,6 +410,15 @@ type BufferOption interface {
 	setBufferOpt(bo *bufferOpts)
 }
 
+// IntersectsOption is an option passed to Geometry.Intersects()
+//
+//
+// Available options are:
+//  - ErrLogger
+type IntersectsOption interface {
+	setIntersectsOpt(bo *bufferOpts)
```
Comment (tbonfort) [inline options.go:419]: you should add and use a new `intersectsOpts` structure, instead of re-using `bufferOpts`

**Review by tbonfort — CHANGES_REQUESTED** [review]

File: godal.cpp (hunk 1 above)
Comment (CodingMarmot) [reply godal.cpp:689]: `OGR_G_Intersects` only returns an `int`, which is `TRUE` or `FALSE`.
It does not return an handle which can be `nullptr`, so we can't have the same error handling.

Or do I miss something?

File: godal.cpp (hunk 2 above)
Comment (CodingMarmot) [reply godal.cpp:862]: Done for next commit.

File: options.go (hunk 3 above)
Comment (CodingMarmot) [reply options.go:419]: Copy paste error, done for next commit.

**Review by tbonfort — APPROVED** [review]

**Pull request merged**